01-Alex-Nicolaidis/wk01/4-thu/bank/js/bank.js

```diff
@@ -0,0 +1,132 @@
+// # JavaScript Bank
+
+// In this homework, you'll create a basic `bank` in Javascript. The bank has many `accounts` and the following capabilities that you need to write.
+// #### Bank
+// There is only one bank. This bank has an array of accounts. The bank needs a method that will return the total sum of money in the accounts. It also needs an `addAccount` method that will enroll a new account at the bank and add it to the array of accounts. There is no need to create additional functions of the bank to delete accounts, etc.
+// The bank has many accounts. Accounts should be objects that all share a set of common functionality. 
+// #### Accounts
+// Accounts have a current balance and owner's name. You should be able to deposit or withdraw from an account to change the balance. 
+// There is no need to write a user interface. Make sure functions return values -- you may also have your functions `console.log()` values to help you see your code working. 
+// You should write a basic story through a series of JavaScript commands that shows that the methods do indeed work as expected: add some accounts, show the total balance, make some deposits and withdrawals, show the new total balance. 
+// ### Tips
+// Don't overthink this. Shorter code is probably the answer.
+// ## Bonus
+// - Ensure that the accounts cannot have negative values. 
+// - Write a 'transfer' on the bank that allows you to transfer amounts between two accounts. 
+// ## Additional
+// Begin exploring the [JavaScript Koans](https://github.com/liammclennan/JavaScript-Koans). Fork, clone and start trying them.
+
+
+
+// FUNCTIONS FOR THE BANK 
+
+let totalBankMoney = function (bank){
+    let totalMoney = 0;
+    for (let i = 0; i < bank.length; i++){
+        let account = bank[i];
+        totalMoney += account.currentBalance;
+    }
+    return totalMoney;
+};
+
+//add an account as an object and push into the banks list of accounts
+const addAccount = function (customerName, initalBalance, bank){
+    const customerAccount = {
+        name: customerName,
+        currentBalance: initalBalance
+    };
+    bank.push(customerAccount);
+};
+
+const withdraw = function (amount, customerName, bank){
+    let withdrawn = amount;
+    for (let i = 0; i < bank.length; i++){
+        const account = bank[i]
+        if ( account.name === customerName ){
+            if( account.currentBalance < amount ){
+                withdrawn = account.currentBalance;
+                console.log(`${account.name} you only have ${withdrawn} left. Here is your $${account.currentBalance} your account is now empty.`);
+                account.currentBalance = 0;
+            }else{
+                account.currentBalance -= amount;
+                console.log(`${account.name} your remaining Balance is $${account.currentBalance}. Here is your $${amount}.`)
+            };
+        };
```

Comment:
```suggestion
            };
            break;
        };
```
Presumably, there will only be one account with any given name. Once you've found it, there's no need to continue looping and so we can use the `break` keyword to end the loop early.